docs/architecture/index.rst

```diff
@@ -1,25 +1,586 @@
+checkpointing, and metric storage.**
+
+The following figure is a more detailed view of the Determined platform:
+
+.. image:: /assets/images/arch11.png
+
+**TBD: Describe what the boxes are and how they interact. And, of course, generalize the diagram.**
+
+************
+ Components
+************
+
+Determined deployment can range from a local laptop and on-premises setups to distributed master and
+agent nodes, all with database and file storage connectivity, containerization, and cloud resource
+options.
+
+The following figure shows the key master and agent instances on separate machines:
+
+.. image:: /assets/images/arch00.png
+
+However, a single machine can have both a master and one agent instance.
+
+Master and Agent Nodes
+======================
+
+Determined comprises a single *master* and one or more *agents* in a cluster environment. The master
+is a single, non-GPU node that manages the cluster.
+
+The master keeps experiment metadata in a PostgreSQL database, which can be queried using the WebUI
+or CLI. You can interact with the master, directly, but cannot interact with agents.
+
+Master Node Scope of Responsibility
+-----------------------------------
+
+Master node responsibilities:
+
+-  Stores experiment, trial, and workload metadata.
+-  Schedules and dispatch experiments to agents.
+-  Dynamically provisions and deprovisions on-premises and cloud agents.
+-  Advances experiment, trial, and workload state machines.
+-  Responds to CLI commands.
+-  Hosts the WebUI, which is primarily used to monitor experiments.
+-  Serves the REST API.
+
+The master is also responsible for the following training-specific searcher, metrics, checkpointing,
+and scheduling functionality:
+
++---------------+----------------------------------------------------------------------+
+| Function      | Description                                                          |
++===============+======================================================================+
+| Searcher      | The *Searcher* implements a hyperparameter search algorithm and is   |
+|               | responsible for coordinating the work of all experiment trials.      |
++---------------+----------------------------------------------------------------------+
+| Metrics       | *Metrics* provides persistent storage of metrics reported by Trials. |
++---------------+----------------------------------------------------------------------+
+| Checkpointing | *Checkpointing* captures the training state at a given time in the   |
+|               | *model registry*.                                                    |
++---------------+----------------------------------------------------------------------+
+| Scheduling    | *Scheduling* schedules jobs to run, ensuring that all of the compute |
+|               | resources required for a job are available before the job launches.  |
++---------------+----------------------------------------------------------------------+
```

Comment:
Only searcher and metrics are training-specific.

Ok, technically, checkpointing is also training-specific, but it shouldn't be, and in the near-to-mid-term future we would like checkpointing to not be training specific.

Might I suggest converting this table into additional bullets?

Additionally, checkpointing does not necessarily put things in the model registry.  Checkpoints are just snapshots of training state.  The model registry is like a key-value store on top of the checkpoints in the system.  There may be many checkpoints in the system but maybe only a small number of them get put (by the user) into the model registry.

Reply:
stylistically, the bullet format is not used for a definition-type construct. Not sure why bullets would better address that checkpointing and scheduling are not training-specific.